### PR TITLE
[Maintain][Convolution] Promote conv1d ops to implemented

### LINKED ISSUE #1555: [MAINTAIN][CONVOLUTION] promote conv1d ops to implemented
### Symptom / Motivation
`Conv1dFwdOp` and `Conv1dBiasFwdOp` have functional kernel/op implementations and targeted correctness coverage, but both manifest entries remain `status: spec-only`. Strict manifest validation still reports missing op-level parity methods, so the entries cannot be promoted to `implemented` yet.

### Root Cause Analysis
The Conv1d op classes in `tileops/ops/convolution.py` do not implement the manifest-required `_validate_dtypes` and `eval_roofline` methods. `_infer_output_shapes` is also not overridden, and the manifest status comments are stale after dilation, groups, and string padding support landed.

### Related Files
- `tileops/manifest/convolution.yaml`
- `tileops/ops/convolution.py`
- `tests/ops/test_convolution.py`
- `benchmarks/ops/bench_convolution.py`
- `scripts/validate_manifest.py`

## Goal
Promote `Conv1dFwdOp` and `Conv1dBiasFwdOp` to `implemented` by making their op implementation pass strict manifest validation and existing Conv1d correctness tests.

## Plan

1. Add Conv1d output-shape, dtype, and roofline methods matching the manifest contract.
2. Update the two Conv1d manifest entries from `spec-only` to `implemented` once strict validation passes.
3. Run targeted Conv1d tests and strict manifest validation for both op entries.

## Constraints
- Keep benchmark manifest-driven migration out of scope because `source.bench_manifest_driven` is currently false and L4 is advisory for these entries.
- Do not change Conv1d public input semantics beyond the manifest-declared contract.

## Acceptance Criteria
- [ ] `python scripts/validate_manifest.py --strict --check-op Conv1dFwdOp` passes
- [ ] `python scripts/validate_manifest.py --strict --check-op Conv1dBiasFwdOp` passes
- [ ] `PYTHONPATH="$PWD" python -m pytest -q tests/ops/test_convolution.py -k "conv1d"` passes
- [ ] `Conv1dFwdOp` and `Conv1dBiasFwdOp` manifest entries are set to `implemented`

### PR DESCRIPTION
**Pull request by RMLYC**
Closes #1555

## Summary

- Add Conv1d output-shape inference, dtype validation, and roofline methods for no-bias and bias variants.
- Validate Conv1d dtypes in `forward()` before dispatching kernels.
- Promote `Conv1dFwdOp` and `Conv1dBiasFwdOp` manifest entries from `spec-only` to `implemented`.

## Test plan

- [x] `python scripts/validate_manifest.py --strict --check-op Conv1dFwdOp`
- [x] `python scripts/validate_manifest.py --strict --check-op Conv1dBiasFwdOp`
- [x] `PYTHONPATH="$PWD" python -m pytest -q tests/ops/test_convolution.py -k "conv1d"`
- [x] `python scripts/validate_manifest.py --strict`
- [x] `PYTHONPATH="$PWD" python -m pytest -q tests/ops/test_convolution.py`
- [x] `git diff --check`
- [x] `python -m compileall -q tileops/ops/convolution.py`
- [x] Commit/push hooks passed

## Regression

- Conv1d targeted tests: `19 passed, 24 deselected`
- Full convolution test file: `43 passed`
- Manifest strict validation passes for both Conv1d entries; remaining Conv1d L4 benchmark manifest-driven messages are advisory because `source.bench_manifest_driven` remains false.

## Additional context

- `scripts/validate.sh` is not present in this checkout, so the PR lifecycle pre/post gates from the skill could not be run directly. The repo-available manifest, pytest, diff, compile, and hook checks above were used instead.

### CODE REVIEW

File: tileops/ops/convolution.py [hunk 1]
```diff
@@ -327,6 +396,47 @@ def forward(
...
+        flops = (
+            2 * self.n * self.c_out * l_out * self.c_in_g * self.kernel_size
+            + self.n * self.c_out * l_out
+        )
+        bytes_ = (
+            self.n * self.c_in * self.l_in
+            + self.c_out * self.c_in_g * self.kernel_size
+            + self.c_out
+            + self.n * self.c_out * l_out
+        ) * self.dtype.itemsize
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py]: Fixed in 36ee448 by replacing the Conv1dBiasFwdOp roofline byte-size calculation with torch.tensor([], dtype=self.dtype).element_size(). Re-ran strict Conv1d manifest validation and targeted Conv1d tests.

File: tileops/ops/convolution.py [hunk 2]
```diff
@@ -266,6 +288,52 @@ def forward(
...
+            self.stride,
+            self.padding,
+            self.dilation,
+        )
+        flops = 2 * self.n * self.c_out * l_out * self.c_in_g * self.kernel_size
+        bytes_ = (
+            self.n * self.c_in * self.l_in
+            + self.c_out * self.c_in_g * self.kernel_size
+            + self.n * self.c_out * l_out
+        ) * self.dtype.itemsize
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py]: Fixed in 36ee448 by replacing the Conv1dFwdOp roofline byte-size calculation with torch.tensor([], dtype=self.dtype).element_size(). Re-ran strict Conv1d manifest validation and targeted Conv1d tests.

File: tileops/manifest/convolution.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ Conv1dFwdOp:
   ref_api: "torch.nn.functional.conv1d"
   # Equivalent to torch.nn.functional.conv1d (bias=None)
   family: convolution
-  status: spec-only  # TODO: impl lacks dilation, groups, string padding
+  status: implemented
```
Comment (superAngGao) [inline tileops/manifest/convolution.yaml:11]: Now that `Conv1dFwdOp` / `Conv1dBiasFwdOp` are promoted to `implemented`, the manifest contract is wider than the implementation for `padding="same"`. The manifest only restricts `same` to `stride == 1`, but `_conv_padding_to_tuple()` also rejects even effective kernels because the current kernel only supports symmetric padding. For example, `Conv1dFwdOp(..., kernel_size=2, padding="same")` raises `ValueError`, while `torch.nn.functional.conv1d(..., padding="same")` supports it via asymmetric padding. Please either implement asymmetric `same` padding, or narrow both Conv1d manifest entries with an explicit odd-effective-kernel rule / contract note, e.g. `_p != "same" or (_d * (kW - 1) + 1) % 2 == 1`.

**Review by superAngGao — CHANGES_REQUESTED** [review]
Thanks for pushing Conv1d toward the implemented manifest contract. I found one contract gap around `padding="same"` that should be addressed before this is marked implemented.

File: tileops/manifest/convolution.yaml (hunk 3 above)
Comment (RMLYC) [reply tileops/manifest/convolution.yaml:11]: Implemented asymmetric `padding="same"` for Conv1d instead of narrowing the manifest contract.

Summary of the fix:
- `Conv1dFwdOp` / `Conv1dBiasFwdOp` keep their public interface unchanged.
- Conv1d now computes concrete `(pad_left, pad_right)` values for `padding="same"`, including even effective kernels.
- `Conv1dKernel` / `GroupConv1dKernel` now carry `pad_l` as `(left, right)` internally and pass `pad_left` / `pad_right` into the lower TileLang kernels.
- The lower Conv1d kernels compute `out_l` from `pad_left + pad_right` and use `pad_left` for input indexing, matching PyTorch asymmetric same-padding behavior.
- `_validate_conv_params` now validates asymmetric padding entries using `pad_left + pad_right` instead of assuming `2 * pad`.
- Added coverage for `kernel_size=2, padding="same"` against `torch.nn.functional.conv1d(..., padding="same")` for both bias and no-bias variants.

Validation passed:
- `python -m compileall -q tileops/ops/convolution.py tileops/kernels/convolution.py tests/ops/test_convolution.py`
- `TILELANG_CLEANUP_TEMP_FILES=1 PYTHONPATH="$PWD" python -m pytest -q tests/ops/test_convolution.py -k "conv1d"`
- `TILELANG_CLEANUP_TEMP_FILES=1 PYTHONPATH="$PWD" python -m pytest -q tests/ops/test_convolution.py`
- `python scripts/validate_manifest.py --strict --check-op Conv1dFwdOp`
- `python scripts/validate_manifest.py --strict --check-op Conv1dBiasFwdOp`

**Comment by superAngGao** [comment]
LGTM, approving. I reviewed the conv1d promote-to-implemented path and the padding="same" asymmetric padding plumbing; the op/kernel/fake-meta flow looks consistent. I also verified strict manifest validation and the conv1d tests locally.

Non-blocking: the manifest shape_rules still use `_s/_p/_d` helper-style expressions that the validator can only warn/skip during parity checks, and the conv1d benchmark remains `bench_manifest_driven: false`. This is fine for this PR, but worth cleaning up in a follow-up to make the implemented status better covered.

**Review by superAngGao — APPROVED** [review]
